### PR TITLE
Edits_on_machine_learning_modules

### PR DESCRIPTION
**Pull request by guanyu-certifai**
Edits on:
01-0_NumPy Quickstart
01-1_Linear Algebra with NumPy
08-0_K Nearest Neighbour(KNN)
10-0_Principal Component Analysis (PCA)

### CODE REVIEW

**Comment by kenghooi-teoh** [comment]
**KNN**
- we always perform normalization **after** splitting the dataset (please look up the reason if you are not sure why)
- compare the evalaution metrics between train vs test
- check the precision/recall too since the glass dataset is imbalanced (163 window vs 51 non-window)

**PCA**
- stick to Python convention for naming vairables is snake-case, e.g. _first_second_
```
principalComponents = pca.fit_transform(scaled_data)
principaldf = pd.DataFrame(data=principalComponents,
                           columns=['Principal component 1', 'Principal component 2'])
principaldf.head()
```
- add a simple note about interpretability of the principal components ( we are not sure what the principal components are representing)

**Linear Algebra with NumPy**
- add a few methods that describe tensor properties such as rank, shape, and examples of higher dimensional tensors
